pkg/subctl/cmd/diagnose/globalnet.go

```diff
@@ -0,0 +1,254 @@
+	if condition == nil {
+		status.Failure("ClusterGlobalEgressIP %q is missing the %q status condition", clusterGlobalEgressIP.Name,
+			submarinerv1.GlobalEgressIPAllocated)
+
+		retValue = false
+	} else if condition.Status != metav1.ConditionTrue {
+		status.Failure("The allocation of global IPs for ClusterGlobalEgressIP %q failed with reason %q and message %q",
+			clusterGlobalEgressIP.Name, condition.Reason, condition.Message)
+
+		retValue = false
+	}
+
+	return retValue
+}
+
+func checkGlobalEgressIps(cluster *cmd.Cluster, status reporter.Interface) bool {
+	globalEgressIps, err := cluster.SubmClient.SubmarinerV1().GlobalEgressIPs(
+		corev1.NamespaceAll).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		status.Warning("Error obtaining GlobalEgressIPs resources: %v", err)
+		return false
+	}
+
+	retValue := true
+
+	for i := range globalEgressIps.Items {
+		gip := globalEgressIps.Items[i]
+		numberOfIPs := -1
+
+		if gip.Spec.NumberOfIPs != nil {
+			numberOfIPs = *gip.Spec.NumberOfIPs
+		}
+
+		if numberOfIPs != len(gip.Status.AllocatedIPs) {
+			status.Failure("The number of requested IPs (%d) does not match the number allocated (%d) for GlobalEgressIP %q",
+				numberOfIPs, len(gip.Status.AllocatedIPs), gip.Name)
+
+			retValue = false
+		}
+	}
+
+	return retValue
+}
+
+func checkGlobalIngressIps(cluster *cmd.Cluster, status reporter.Interface) bool {
+	mcsClient, err := mcsClientset.NewForConfig(cluster.Config)
+	if err != nil {
+		status.Warning("Error obtaining mcs client: %v", err)
+		return false
+	}
+
+	serviceExports, err := mcsClient.MulticlusterV1alpha1().ServiceExports(corev1.NamespaceAll).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		status.Warning("Error listing ServiceExport resources: %v", err)
+		return false
+	}
+
+	retValue := true
+
+	for i := range serviceExports.Items {
+		ns := serviceExports.Items[i].GetNamespace()
+		name := serviceExports.Items[i].GetName()
+
+		svc, err := cluster.KubeClient.CoreV1().Services(ns).Get(context.TODO(), name, metav1.GetOptions{})
+
+		if apierrors.IsNotFound(err) {
+			status.Warning("No matching Service resource found for exported service \"%s/%s\"", ns, name)
+			continue
+		}
+
+		if err != nil {
+			status.Failure("Error retrieving Service \"%s/%s\", %v", ns, name, err)
+
+			retValue = false
+
+			continue
+		}
+
+		if svc.Spec.Type != corev1.ServiceTypeClusterIP {
+			continue
+		}
+
+		globalIngress, err := cluster.SubmClient.SubmarinerV1().GlobalIngressIPs(ns).Get(context.TODO(), name, metav1.GetOptions{})
+
+		if apierrors.IsNotFound(err) {
+			status.Failure("No matching GlobalIngressIP resource found for exported service \"%s/%s\"", ns, name)
+
+			retValue = false
+
+			continue
+		}
+
+		if err != nil {
+			status.Failure("Error retrieving GlobalIngressIP for exported service \"%s/%s\": %v", ns, name, err)
+			return false
+		}
+
```

Comment:
Check the status condition here.

Reply:
Not sure that I understood this comment

Reply:
I mean the GlobalIngressIP status Condition as was done with the *EgressIPs (c. L119).